api/clients/eigenda_client.go

```diff
@@ -265,3 +301,33 @@ func (m *EigenDAClient) putBlob(ctx context.Context, rawData []byte, resultChan
 func (c *EigenDAClient) Close() error {
 	return c.Client.Close()
 }
+
+// getConfDeepBlockNumber returns the block number that is `depth` blocks behind the current block number.
+func (m EigenDAClient) getConfDeepBlockNumber(ctx context.Context, depth uint64) (*big.Int, error) {
+	curBlockNumber, err := m.ethClient.BlockNumber(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get latest block number: %w", err)
+	}
+	// If curBlock < depth, this will return the genesis block number (0),
+	// which would cause to accept as confirmed a block that isn't depth deep.
+	// TODO: there's prob a better way to deal with this, like returning a special error
+	return new(big.Int).SetUint64(max(curBlockNumber-depth, 0)), nil
```

Comment:
nix, since depth is uint64, logically, it can underflow

Reply:
good find: [32a6095](https://github.com/Layr-Labs/eigenda/pull/821/commits/32a60954433a474ce12fc82f7be77d67949dee13)